src/Catalyst.Core.Modules.Dfs/UnixFs/SizeChunker.cs

```diff
@@ -0,0 +1,199 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Abstractions.Options;
+using Catalyst.Core.Lib.Dag;
+using ProtoBuf;
+
+namespace Catalyst.Core.Modules.Dfs.UnixFileSystem
+{
+    /// <summary>
+    ///     Chunks a data stream into data blocks based upon a size.
+    /// </summary>
+    public class SizeChunker
+    {
+        /// <summary>
+        ///     Performs the chunking.
+        /// </summary>
+        /// <param name="stream">
+        ///     The data source.
+        /// </param>
+        /// <param name="name">
+        ///     A name for the data.
+        /// </param>
+        /// <param name="options">
+        ///     The options when adding data to the IPFS file system.
+        /// </param>
+        /// <param name="blockService">
+        ///     The destination for the chunked data block(s).
+        /// </param>
+        /// <param name="keyChain">
+        ///     Used to protect the chunked data blocks(s).
+        /// </param>
+        /// <param name="cancel">
+        ///     Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException" /> is raised.
+        /// </param>
+        /// <returns>
+        ///     A task that represents the asynchronous operation. The task's value is
+        ///     the sequence of file system nodes of the added data blocks.
+        /// </returns>
+        public async Task<List<UnixFsNode>> ChunkAsync(Stream stream,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)